### PR TITLE
feat: Check if federation is enabled before fetching a conversation

### PR DESCRIPTION
**Pull request by KaterinaWire**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Since we use fetch(with remoteIdentifier: UUID, domain: String?, in context: NSManagedObjectContext) -> ZMConversation? method in other frameworks, it makes sense to check if federation is enabled inside it.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by David-Henner — APPROVED** [review]

File: Tests/Source/Model/ZMManagedObjectFetchingTests.swift [hunk 1]
```diff
@@ -26,6 +27,7 @@ class ZMManagedObjectFetchingTests: DatabaseBaseTest {
         super.setUp()
         self.mocs = self.createStorageStackAndWaitForCompletion()
         XCTAssert(waitForAllGroupsToBeEmpty(withTimeout: 1))
+        APIVersion.isFederationEnabled = true
```
Comment (johnxnguyen) [inline Tests/Source/Model/ZMManagedObjectFetchingTests.swift:30]: It would be good to reset the value back to false in the tear down method.

**Review by johnxnguyen — CHANGES_REQUESTED** [review]
It would also be good to have the same logic when fetching `ZMUser`.

File: Tests/Source/Model/ZMManagedObjectFetchingTests.swift (hunk 1 above)
Comment (KaterinaWire) [reply Tests/Source/Model/ZMManagedObjectFetchingTests.swift:30]: Done

**Comment by KaterinaWire** [comment]
> It would also be good to have the same logic when fetching `ZMUser`.

ZMUser.fetchOrCreate(with remoteIdentifier: UUID,...) method uses the same method:
```
@objc static func fetchOrCreate(with remoteIdentifier: UUID,
                                    domain: String?,
                                    in context: NSManagedObjectContext,
                                    created: UnsafeMutablePointer<Bool>) -> ZMUser {
        // We must only ever call this on the sync context. Otherwise, there's a race condition
        // where the UI and sync contexts could both insert the same user (same UUID) and we'd end up
        // having two duplicates of that user, and we'd have a really hard time recovering from that.
        require(context.zm_isSyncContext, "Users are only allowed to be created on sync context")

        let domain: String? = APIVersion.isFederationEnabled ? domain : nil

        if let user = fetch(with: remoteIdentifier, domain: domain, in: context) {
            return user
        } else {
            created.pointee = true
            let user = ZMUser.insertNewObject(in: context)
            user.remoteIdentifier = remoteIdentifier
            user.domain = domain?.selfOrNilIfEmpty
            return user
        }
    }
```
The changes in this PR are related to the `ZMManagedObject`  extension. Sorry, the PR description didn't explain that this is also related to `ZMUser`.

**Review by johnxnguyen — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release patch** for PR #**1362**. Will create a **patch** release after merged.
Next version is 287.2.1.

**Pull request merged**